src/test/java/map/BattleMapTest.java

```diff
@@ -44,12 +44,161 @@ void allFieldsInAnEmptyBattlefieldShouldBeZeros() {
   }
 
   @Test
-  @DisplayName("In coordinate p the number must be 1")
-  void checkSpecificPosition() {
-    BattleMap battleMap = new BattleMap(10);
-    Point p = new Point(10, 10);
+  @DisplayName("Test on changing the cell state from Healthy to Free")
```

Comment:
The test description does not reflect what is going on in the test. We cannot infer from the description, whether the state change is allowed or not but the code shows us that the operation should fail.